crypto/evp/signature.c

```diff
@@ -664,12 +939,88 @@ int EVP_PKEY_sign(EVP_PKEY_CTX *ctx,
 
 int EVP_PKEY_verify_init(EVP_PKEY_CTX *ctx)
 {
-    return evp_pkey_signature_init(ctx, EVP_PKEY_OP_VERIFY, NULL);
+    return evp_pkey_signature_init(ctx, NULL, EVP_PKEY_OP_VERIFY, NULL);
 }
 
 int EVP_PKEY_verify_init_ex(EVP_PKEY_CTX *ctx, const OSSL_PARAM params[])
 {
-    return evp_pkey_signature_init(ctx, EVP_PKEY_OP_VERIFY, params);
+    return evp_pkey_signature_init(ctx, NULL, EVP_PKEY_OP_VERIFY, params);
+}
+
+int EVP_PKEY_verify_init_ex2(EVP_PKEY_CTX *ctx,
+                             EVP_SIGNATURE *algo, const OSSL_PARAM params[])
+{
+    return evp_pkey_signature_init(ctx, algo, EVP_PKEY_OP_VERIFY, params);
+}
+
+int EVP_PKEY_verify_message_init(EVP_PKEY_CTX *ctx,
+                                 EVP_SIGNATURE *algo, const OSSL_PARAM params[])
+{
+    return evp_pkey_signature_init(ctx, algo, EVP_PKEY_OP_VERIFYMSG, params);
+}
+
+int EVP_PKEY_CTX_set_signature(EVP_PKEY_CTX *ctx,
```

Comment:
Are the committers that approved this happy with this API that needs to copy the signature? Bear in mind that signatures are getting bigger.  Something like LMS can have quite a bit of info in the signature.

Reply:
(i.e.  A release is being done soon - I dont want new API's going into 3.4 if we cant change them later).

Reply:
(Dont know why I bother :))

Reply:
I did ask questions on this about a month ago, see https://github.com/openssl/openssl/pull/22672#discussion_r1685961220
Those questions are yet unanswered.

At this point, I'd suggest raising an issue on this, or a PR.

Reply:
As long as we can pass a param in the future that is a ptr I guess we can deal with it later and then have to handle both cases.

Reply:
Sure, leave it 'til later sounds ok to me